ci/openstack/provision.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+set -euo pipefail
+
+source ci/openstack/vars.sh
+if [ "${RUN_OPENSTACK_CI:-}" != "true" ]; then
+    echo RUN_OPENSTACK_CI is set to false, skipping the openstack end to end test.
+    exit
+fi
+
+if [ "${CI_OVER_CAPACITY:-}" == "true" ]; then
+    echo the CI is over capacity, skipping the end-end test.
+    exit 1
+fi
+
+echo "Keypair name: $KEYPAIR_NAME"
+
+openstack keypair create "$KEYPAIR_NAME" > ~/.ssh/id_rsa
+chmod 600 ~/.ssh/id_rsa
+
+echo CONFIGURE THE INVENTORY
+
+export INVENTORY="$PWD/playbooks/provisioning/openstack/sample-inventory"
+
+mv "$INVENTORY"/ansible.cfg .
+
+PUBLIC_IP="$(curl --silent https://api.ipify.org)"
+
+
+cat << EOF >> extra-vars.yaml
+openstack_ssh_public_key: $KEYPAIR_NAME
+openstack_external_network_name: "38.145.32.0/22"
+openstack_default_image_name: "CentOS-7-x86_64-GenericCloud-1703"
+openstack_num_nodes: 1
+env_id: $ENV_ID
+node_ingress_cidr: "$PUBLIC_IP/32"
+ssh_ingress_cidr: "$PUBLIC_IP/32"
+manage_packages: False
+ephemeral_volumes: True
+EOF
+
+
+# TODO(shadower): this just started to break. We will need to unset this from
+# the sample inventory.
+sed -i 's/openshift_release/#openshift_release/' "$INVENTORY"/group_vars/OSEv3.yml
+
+cat << EOF >> "$INVENTORY"/group_vars/OSEv3.yml
+openshift_master_identity_providers:
```

Comment:
these may be as well specified as extra-vars, the latter takes precedence over the group vars